### PR TITLE
feat: add azure-identity node example

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

part of https://github.com/Azure/azure-workload-identity/issues/187

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: examples/azure-identity/node/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ARG BUILDER=node:14
```
Comment (sozercan) [inline examples/azure-identity/node/Dockerfile:1]:
```suggestion
ARG BUILDER=mcr.microsoft.com/mirror/docker/library/node:16
```

File: examples/azure-identity/node/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ARG BASEIMAGE=gcr.io/distroless/nodejs:16
```
Comment (sozercan) [inline examples/azure-identity/node/Dockerfile:2]:
```suggestion
ARG BASEIMAGE=mcr.microsoft.com/cbl-mariner/base/nodejs:16
```

File: examples/azure-identity/node/Dockerfile (hunk 1 above)
Comment (aramase) [reply examples/azure-identity/node/Dockerfile]: updated in this and [here](https://github.com/Azure/azure-workload-identity/pull/580/files#diff-aec6fad245e042e4b50d8431d13942881c29de6060f9475485e292e2bada551f)

File: examples/azure-identity/node/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+ARG BUILDER=mcr.microsoft.com/cbl-mariner/base/nodejs:16
+ARG BASEIMAGE=gcr.io/distroless/nodejs:16
```
Comment (sozercan) [inline examples/azure-identity/node/Dockerfile:2]:
```suggestion
ARG BASEIMAGE=mcr.microsoft.com/mirror/gcr/distroless/nodejs-debian11:16
```

File: examples/msal-node/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ARG BUILDER=node:14
+ARG BUILDER=mcr.microsoft.com/cbl-mariner/base/nodejs:16
 ARG BASEIMAGE=gcr.io/distroless/nodejs:16
```
Comment (sozercan) [inline examples/msal-node/Dockerfile:2]:
```suggestion
ARG BASEIMAGE=mcr.microsoft.com/mirror/gcr/distroless/nodejs-debian11:16
```

**Review by sozercan — APPROVED** [review]

**Pull request merged**